lib/bibliothecary/multi_parsers/spdx_json.rb

```diff
@@ -0,0 +1,80 @@
+require "package_url"
+
+module Bibliothecary
+  module MultiParsers
+    module SpdxJson
+      include Bibliothecary::Analyser
+      include Bibliothecary::Analyser::TryCache
+
+      NoEntries = Class.new(StandardError)
+
+      def self.mapping
+        {
+          match_extension(".json") => {
+            content_matcher: :spdx_json_file?,
+            kind: "lockfile",
+            parser: :parse_spdx_json,
+            ungroupable: true,
+          },
+        }
+      end
+
+      def spdx_json_file?(file_contents)
+        manifest = JSON.parse(file_contents)
+        spdx_id = manifest.fetch("SPDXID")
+        spdx_version = manifest.fetch("spdxVersion")
+        return spdx_id == "SPDXRef-DOCUMENT" && !spdx_version.nil?
+      rescue Exception # rubocop:disable Lint/RescueException
+        # We rescue exception here since native libs can throw a non-StandardError
+        # We don't want to throw errors during the matching phase, only during
+        # parsing after we match.
+        false
+      end
+
+      def parse_spdx_json(file_contents, options: {})
+        entries = try_cache(options, options[:filename]) do
+          parse_spdx_json_file_contents(file_contents)
+        end
+
+        raise NoEntries if entries.empty?
+
+        entries[platform_name.to_sym]
+      end
+
+      def get_platform(purl_string)
+        platform = PackageURL.parse(purl_string).type
+
+        Bibliothecary::PURL_TYPE_MAPPING[platform]
+      end
+
+      def parse_spdx_json_file_contents(file_contents)
+        entries = {}
+        manifest = JSON.parse(file_contents)
+
+        manifest["packages"]&.each do |package|
+          package_name = package["name"]
+          package_version = package["versionInfo"]
+
+          platform = nil
+          package["externalRefs"]&.each do |ref|
+            if ref["referenceType"] == "purl"
+              purl = ref["referenceLocator"]
+              platform ||= get_platform(purl)
```

Comment:
I feel like if we have a purl, we might get a better package name from it than we do from the name field. At least that seemed to be the case when I briefly looked a week or two ago

Reply:
Yeah, we definitely want to use the name from the purl if there is one. Looking at an example spdx json sbom generated from github has 
```
    {
      "SPDXID": "SPDXRef-nuget-AWSCRT-AUTH-0.4.4",
      "name": "nuget:AWSCRT-AUTH",
      "versionInfo": "0.4.4",
      "downloadLocation": "NOASSERTION",
      "filesAnalyzed": false,
      "licenseConcluded": "Apache-2.0",
      "supplier": "NOASSERTION",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:nuget/AWSCRT-AUTH@0.4.4",
          "referenceType": "purl"
        }
      ]
    },
```

The name has the platform embedded in it but there's no real structure to what the name is supposed to be that I can see. So I think if there's a purl, we want to use the information out of that and then fallback to the non-purl fields

Reply:
Sounds like that should also apply to the tag:value spdx parser. I'll update both.